### ISSUE TITLE
Documentation for "char_rnn_classification_tutorial" has wrong output cell calculation

### ISSUE THREAD

**Issue by williamFalcon** [pytorch/pytorch#4573]
@soumith Looks like [this tutorial](http://pytorch.org/tutorials/intermediate/char_rnn_classification_tutorial.html) has a bug in this code:      

```python   
import torch.nn as nn
from torch.autograd import Variable

class RNN(nn.Module):
    def __init__(self, input_size, hidden_size, output_size):
        super(RNN, self).__init__()

        self.hidden_size = hidden_size

        self.i2h = nn.Linear(input_size + hidden_size, hidden_size)
        self.i2o = nn.Linear(input_size + hidden_size, output_size)
        self.softmax = nn.LogSoftmax(dim=1)

    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(combined)
        output = self.softmax(output)
        return output, hidden

    def initHidden(self):
        return Variable(torch.zeros(1, self.hidden_size))

n_hidden = 128
rnn = RNN(n_letters, n_hidden, n_categories)
```    

Looks like the output of the forward pass is done on the input and NOT on the processed data as computed by the RNN cell (ie: self.i2h(combined))   

So, this:    
```python    
    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(combined)
        output = self.softmax(output)
        return output, hidden
```   
Should become:    
```python    
    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(hidden) # this line needs to be changed    
        output = self.softmax(output)
        return output, hidden
```
  
If you guys agree, I'm happy to make that minor (but frustrating if you don't know what's going on) change to the docs.

**Comment by apaszke** [comment]
Sorry, but what exactly is the problem? It doesn't matter if you use `hidden` or `combined` - you're composing linear operators, so you can alternatively use a single one, which is what the code does in here. Also, if you want to discuss that please open an issue in pytorch/tutorials repo (that's where they are hosted). Thanks!

**Comment by williamFalcon** [comment]
@apaszke ok will open there. 

The issue is that seld.i2o is intended to map the hidden into classification space (with the softmax).  

However, as it’s written now, it is mapping the input directly without processing in the cell. 

That is, current code:

ht = (xt * W_hx + b) + (ht_prev * W_hh + b)  
output_t = (xt * W_hx + b ) + (ht_prev * W_hh + b)  

But it really should be:
output_t = (ht * W_io + b)
